be/src/vec/data_types/data_type_decimal.h

```diff
@@ -662,7 +710,8 @@ void convert_to_decimal(typename ToDataType::FieldType* dst,
                                                       Decimal256, Decimal64>>;
         convert_to_decimals<DataTypeDecimal<DecimalFrom>, ToDataType, multiply_may_overflow,
                             narrow_integral>(dst, src, from_scale, to_scale, min_result, max_result,
-                                             size);
+                                             size, IsDateV2Type<FromDataType>,
+                                             IsDateTimeV2Type<FromDataType>);
```

Comment:
IsDateTimeV2Type<FromDataType> should be move to function `convert_to_decimals`, and it will be something like
`convert_to_decimals(...) {
if constexpr (IsDateTimeV2Type<>) {
} else if constexpr (IsDateV2Type) {
} else {
   ...
}
}
`